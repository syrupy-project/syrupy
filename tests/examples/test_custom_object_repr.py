class MyCustomClass:
    prop1 = 1
    prop2 = "a"
    prop3 = {1, 2, 3}


def test_snapshot_custom_class(snapshot):
    assert MyCustomClass() == snapshot


class MyCustomReprClass(MyCustomClass):
    def __repr__(self):
        state = "\n".join(
            f"  {a}={repr(getattr(self, a))},"
            for a in sorted(dir(self))
            if not a.startswith("_")
        )
        return f"{self.__class__.__name__}(\n{state}\n)"


def test_snapshot_custom_repr_class(snapshot):
    assert MyCustomReprClass() == snapshot
