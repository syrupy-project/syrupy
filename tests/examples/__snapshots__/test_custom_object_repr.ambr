# name: test_snapshot_custom_class
  <class 'MyCustomClass'> {
    prop1=1,
    prop2='a',
    prop3=<class 'set'> {
      1,
      2,
      3,
    },
  }
---
# name: test_snapshot_custom_repr_class
  MyCustomReprClass(
    prop1=1,
    prop2='a',
    prop3={1, 2, 3},
  )
---
